harvester/utils.py

```diff
@@ -119,11 +113,42 @@ def calculate_batch_size(col_size):
         return 500
 
 
-def split_into_batches(bindings):
+def split_into_download_batches(bindings):
     """
     Split a collection into download batches.
     """
     col_size = len(bindings)
     batch_size = calculate_batch_size(col_size)
     batches = [bindings[i : i + batch_size] for i in range(0, col_size, batch_size)]
     return batches
+
+
+def bindings_with_prefix(bindings, prefix):
+    """
+    Find DC identifiers of which binding ID start with a given prefix.
+    """
+    return [
+        binding
+        for binding in bindings
+        if binding_id_from_dc(binding).startswith(prefix)
+    ]
+
+
+def assign_bindings_to_images(bindings, max_bindings_per_image, shared_prefix=""):
+    """
+    Split a list of bindings into images, each image containing no more than
+    max_bindings_per_image bindings.
+    """
+    images = []
+    for i in range(10):
+        prefix = shared_prefix + str(i)
+        prefixed_bindings = bindings_with_prefix(bindings, prefix)
+        if len(prefixed_bindings) <= max_bindings_per_image:
+            images.append({"prefix": prefix, "bindings": prefixed_bindings})
+        else:
+            images.extend(
+                assign_bindings_to_images(
+                    prefixed_bindings, max_bindings_per_image, prefix
+                )
+            )
+    return images
```

Comment:
re: RC discussion about not splitting small collections at all. We can make them go into a single image by checking whether an image is small enough not after it has been split but before, something like this:
```
    if len(bindings) <= max_bindings_per_image:
        return [{"prefix": shared_prefix, "bindings": bindings}]

    images = []
    for i in range(10):
        prefix = shared_prefix + str(i)
        prefixed_bindings = bindings_with_prefix(bindings, prefix)
        results.extend(
            assign_bindings_to_images(prefixed_bindings, max_bindings_per_image, prefix)
            )
    return images
```